host/frontend/host-orchestrator/operation.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"sync"
+)
+
+type NewOperationError string
+
+func (s NewOperationError) Error() string {
+	return fmt.Sprintf("failure creating a new operation: %s", string(s))
+}
+
+type Operation struct {
+	Name   string
+	Done   bool
+	Result OperationResult
+}
+
+type OperationResult struct {
+	Error OperationResultError
+}
+
+type OperationResultError struct {
+	ErrorMsg string
+}
+
+type OperationManager interface {
+	New() Operation
+
+	Get(string) (Operation, bool)
```

Comment:
Can this return an error instead of a bool? A `nil` error can mean success, and a non-`nil` error is more informative than a boolean.

Reply:
agree, I think error here makes the interface more extensible.